native/src/main/java/org/soap/ExecutionCallback.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.soap;
+
+import io.ballerina.runtime.api.Future;
+import io.ballerina.runtime.api.async.Callback;
+import io.ballerina.runtime.api.values.BError;
+
+/**
+ * Callback class for executing Ballerina dependently type methods.
+ */
+public class ExecutionCallback implements Callback {
+    private final Future future;
+
+    ExecutionCallback(Future future) {
+        this.future = future;
+    }
+    @Override
+    public void notifySuccess(Object o) {
+        this.future.complete(o);
+    }
+
+    @Override
+    public void notifyFailure(BError bError) {
+        this.future.complete(bError);
```

Comment:
If there is a panic, shouldnt we exit the application?

Reply:
Fixed by [74145ed](https://github.com/ballerina-platform/module-ballerina-soap/pull/112/commits/74145ed5edd44abb461d9a099bef12d689d49fbb)